format-delimited/src/main/java/io/cdap/plugin/format/delimited/common/TypeInference.java

```diff
@@ -0,0 +1,209 @@
+   */
+  public static boolean isDate(String value) {
+    return DateTimePattern.isDate(value);
+  }
+
+  /**
+   * Detects if the given value is of a time type.
+   *
+   * @param value The given raw string value.
+   * @return true if the value is of Time data type
+   */
+  public static boolean isTime(String value) {
+    return DateTimePattern.isTime(value);
+  }
+
+  /**
+   * Detects if the given value is blank or null.
+   *
+   * @param value The given raw string value.
+   * @return true if the value is blank or null, false otherwise.
+   */
+  public static boolean isEmpty(String value) {
+    return value == null || value.trim().length() == 0;
+  }
+
+  /**
+   * Investigates whether the detected integer (math) number is of int, long or String data type depending on the
+   * size of the number.
+   *
+   * @param value The given raw string value.
+   * @return The data type detected. Could be one of int, long or string.
+   */
+  private static DataType deepInvestigateInt(String value) {
+    try {
+      try {
+        int parsedIntValue = Integer.parseInt(value);
+        return DataType.INTEGER;
+      } catch (NumberFormatException e) {
+        Long.parseLong(value);
+        return DataType.LONG;
+      }
+    } catch (NumberFormatException e) {
+      return DataType.STRING;
+    }
+  }
+
+  /**
+   * Investigates whether the detected double (math) number is of double or string data type depending on the size of
+   * the number.
+   *
+   * @param value The given raw string value.
+   * @return The data type detected. Could be one of long or string.
+   */
+  private static DataType deepInvestigateDouble(String value) {
+    try {
+      Double doubleValue = Double.parseDouble(value);
+      if (doubleValue.toString().equals(value)) {
+        return DataType.DOUBLE;
+      }
+      return DataType.STRING;
+    } catch (NumberFormatException e) {
+      return DataType.STRING;
+    }
+  }
+
+  /**
+   * Investigates whether the detected integer (math) number is of long or string data type depending on the size of
+   * the number.
+   *
+   * @param value The given raw string value.
+   * @return The data type detected. Could be one of long or string.
+   */
+  private static DataType deepInvestigateLong(String value) {
+    try {
+      Long.parseLong(value.replace("L", ""));
+      return DataType.LONG;
+    } catch (NumberFormatException e) {
+      return DataType.STRING;
+    }
+  }
+
+  /**
+   * Returns the detected data type of the given value.
+   *
+   * @param value The given raw string value.
+   * @return The inferred {@link DataType} of the given value.
+   */
+  public static DataType getDataType(String value) {
+    if (TypeInference.isEmpty(value)) {
+      return DataType.EMPTY;
+    } else if (TypeInference.isBoolean(value)) {
+      return DataType.BOOLEAN;
+    } else if (TypeInference.isInteger(value)) {
+      return deepInvestigateInt(value);
+    } else if (TypeInference.isDouble(value)) {
```

Comment:
shouldn't we check isLong before isDouble (long has higher priority than double)?

Reply:
Ordering here doesn't matter. For the purpose of clarity I've moved `Long` before `Double` detection